### PR TITLE
LMS subscription flow bugs fixes

### PR DESCRIPTION
**Pull request by rehan99000**
**Description:**
Fixes subscription flow bugs.

**List of bugs addressed:**
- Enrollment record not linked with subscription when learner enrolls in course via subscription (https://edlyio.atlassian.net/browse/EDLY-2026)
- Maximum allowed courses is not being recorded
 (https://edlyio.atlassian.net/browse/EDLY-2028)
- User subscription record being created for the wrong user on LMS (https://edlyio.atlassian.net/browse/EDLY-2029)
- A user could only have one subscription package despite being linked to multiple sub sites (https://edlyio.atlassian.net/browse/EDLY-2030)

### CODE REVIEW

**Review by zubair-arbi — APPROVED** [review]
Changes looks good. 👍 
I believe the tests will be added later.

File: openedx/features/subscriptions/models.py [hunk 1]
```diff
@@ -84,16 +86,18 @@ def is_valid(self):
             return self.expiration_date >= date.today() and course_enrollments_count < self.max_allowed_courses
 
     @classmethod
-    def get_valid_subscriptions(self, user_id=None):
+    def get_valid_subscriptions(self, site, user=None):
```
Comment (ziafazal) [inline openedx/features/subscriptions/models.py:89]: nit: replace kwarg `user` with `username`

File: openedx/features/subscriptions/utils.py [hunk 2]
```diff
@@ -4,18 +4,21 @@
...
-def track_subscription_enrollment(subscription_id, username, course_id):
+def track_subscription_enrollment(subscription_id, user, course_id, site):
     """
     Add user enrollment to valid subscription.
     """
     if subscription_id:
-        enrollment = api.get_enrollment(username, course_id)
-        user = User.objects.get(username=username)
-        valid_user_subscription = UserSubscription.get_valid_subscriptions(user.id).first()
+        enrollment = CourseEnrollment.objects.get(
```
Comment (ziafazal) [inline openedx/features/subscriptions/utils.py:15]: `get` without catching `DoesNotExist` exception should be avoided.

**Review by ziafazal — APPROVED** [review]
@rehan99000 overall looks good. couple of nits.

File: openedx/features/subscriptions/utils.py (hunk 2 above)
Comment (rehan99000) [reply openedx/features/subscriptions/utils.py:15]: As this get is called right after the relevant enrollment is created/updated, it should not throw an exception. Regardless, let me put it in try catch

File: openedx/features/subscriptions/models.py (hunk 1 above)
Comment (rehan99000) [reply openedx/features/subscriptions/models.py:89]: done

**Pull request merged**